### PR TITLE
Clean up native auth common logging

### PR DESCRIPTION
**Pull request by Yuki-YuXin**
### Goal:
- Review our log statements and reduce to only those that are relevant and useful. Especially the logMethodCall should be much reduced. Not every method call is relevant to know about when debugging.
- Double check that the existing logs use string interpolation correctly to avoid leaking sensitive information in the logs.

### Changes summary:
- Every log item has correlation argument. (make sure use getCorrelationId() )
- The [NativeAuthMsalController.kt](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/2374/files#diff-304daa7418210ef9cd11ee385364a4a6f74256dc627f404e53422ec50f75b282) layer acts as the main log layer. Instead, remove logSessions under OAuth2Configuration, OAuth2Strategy, RequestProvider since the users care more about the outcome of the request than the construction and sending of the request. Apply the rule "only logger unknown error" to NativeAuthMsalController.
- Remove // TODO 
- Add request and result Logger.infoWithObject in interactors.
- infoWithObject(final String tag, final String correlationID, final String message, final ILoggable object) like warnWithObject

### Company PR:
Previous comments for reference: https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/2374
msal: https://github.com/AzureAD/microsoft-authentication-library-for-android/pull/2071

### CODE REVIEW

File: common/src/main/java/com/microsoft/identity/common/nativeauth/internal/controllers/NativeAuthMsalController.kt [hunk 1]
```diff
@@ -1520,6 +1520,11 @@ class NativeAuthMsalController : BaseNativeAuthController() {
         oAuth2Strategy: NativeAuthOAuth2Strategy,
         parameters: SignUpStartCommandParameters
     ): SignUpStartApiResult {
+        LogSession.logMethodCall(
```
Comment (SammyO) [inline common/src/main/java/com/microsoft/identity/common/nativeauth/internal/controllers/NativeAuthMsalController.kt:1480]: @Yuki-YuXin  given that the aim of this PR was to clean up logging, what's the reason for introducing logs here? Did you find that these are helpful breadcrumbs to have when e.g. debugging issues?

File: common4j/src/main/com/microsoft/identity/common/java/nativeauth/authorities/NativeAuthCIAMAuthority.kt [hunk 2]
```diff
@@ -104,7 +104,6 @@ class NativeAuthCIAMAuthority (
             correlationId = null,
             "${TAG}.getChallengeTypesWithDefault"
         )
-        Logger.info(TAG, "Challenge Types passed = $challengeTypes")
```
Comment (SammyO) [inline common4j/src/main/com/microsoft/identity/common/java/nativeauth/authorities/NativeAuthCIAMAuthority.kt:107]: Why was this removed? I'd say this is helpful debugging information.

File: common4j/src/main/com/microsoft/identity/common/java/nativeauth/providers/NativeAuthResponseHandler.kt [hunk 3]
```diff
@@ -385,7 +385,15 @@ class NativeAuthResponseHandler {
                 response.body,
                 MicrosoftStsTokenResponse::class.java
             )
-            // TODO add safe logging
+
+            Logger.info(TAG, "MicrosoftStsTokenResponse " +
```
Comment (SammyO) [inline common4j/src/main/com/microsoft/identity/common/java/nativeauth/providers/NativeAuthResponseHandler.kt]: Why are we putting this back?

Comment (SammyO) [reply common4j/src/main/com/microsoft/identity/common/java/nativeauth/providers/NativeAuthResponseHandler.kt]: Right, because there's a TODO there. IMO we don't need this log. Does the browser-delegated flow log something like this?

File: common4j/src/main/com/microsoft/identity/common/java/nativeauth/providers/interactors/ResetPasswordInteractor.kt [hunk 4]
```diff
@@ -94,11 +103,29 @@ class ResetPasswordInteractor(
...
+        Logger.infoWithObject(
+            "${TAG}.rawResponseToResetPasswordStartApiResult",
+            apiResponse.correlationId,
+            "rawApiResponse = ",
+            apiResponse
+        )
+
+        val result = apiResponse.toResult()
+
+        Logger.infoWithObject(
```
Comment (SammyO) [inline common4j/src/main/com/microsoft/identity/common/java/nativeauth/providers/interactors/ResetPasswordInteractor.kt:121]: This is a good idea. Did you test this? Does this not produce too many logs? If you're fine with this then so am I.

File: common4j/src/main/com/microsoft/identity/common/java/nativeauth/authorities/NativeAuthCIAMAuthority.kt (hunk 2 above)
Comment (Yuki-YuXin) [reply common4j/src/main/com/microsoft/identity/common/java/nativeauth/authorities/NativeAuthCIAMAuthority.kt:107]:
```kotlin
val challengeTypesWithDefault = (challengeTypes ?: emptyList()).plus(listOf(NativeAuthConstants.GrantType.REDIRECT)).distinct().joinToString(" ")
Logger.info(TAG, "Challenge Types used = $challengeTypesWithDefault")
return challengeTypesWithDefault
```
Put it back and use challengeTypesWithDefault

File: common/src/main/java/com/microsoft/identity/common/nativeauth/internal/controllers/NativeAuthMsalController.kt (hunk 1 above)
Comment (Yuki-YuXin) [reply common/src/main/java/com/microsoft/identity/common/nativeauth/internal/controllers/NativeAuthMsalController.kt:1480]: Here are some comments and discussion before in a legacy PR. https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/2374
One of the comments was raising an idea that remove the logs from OAuth2Strategy, RequestProvider and RequestHandler, but have some logging in the API layer.

Comment (SammyO) [reply common/src/main/java/com/microsoft/identity/common/nativeauth/internal/controllers/NativeAuthMsalController.kt:1480]: Gotcha, thanks for reminder me. All good.

File: common4j/src/main/com/microsoft/identity/common/java/nativeauth/providers/NativeAuthResponseHandler.kt (hunk 3 above)
Comment (Yuki-YuXin) [reply common4j/src/main/com/microsoft/identity/common/java/nativeauth/providers/NativeAuthResponseHandler.kt]: The browser-delegated flow log something like this:

 In terms of the TokenResponse of MSAL:
MicrosoftStsOAuth2Strategy:TokenResult:MicrosoftStsTokenResponse INFO {"ext_expires_in":4165,"expires_in":4165,"mResponseReceivedTime":0,"scope":"email openid profile Application.Read.All User.Read","token_type":"Bearer"}

BaseController:TokenResult:MicrosoftStsTokenResponse INFO {"ext_expires_in":4165,"mRefreshTokenAge":"","mSpeRing":"","expires_in":4165,"mResponseReceivedTime":0,"scope":"email openid profile Application.Read.All User.Read","token_type":"Bearer"}

Thus update the contents of the log to include what MSAL displays. But we can remove this log as well because token seems very risky and sensitive to me.

File: common4j/src/main/com/microsoft/identity/common/java/nativeauth/providers/interactors/ResetPasswordInteractor.kt (hunk 4 above)
Comment (Yuki-YuXin) [reply common4j/src/main/com/microsoft/identity/common/java/nativeauth/providers/interactors/ResetPasswordInteractor.kt:121]: I think this change also comes from the idea that remove the logs from OAuth2Strategy, RequestProvider and RequestHandler, but have some logging in the API layer. 
In my opinion and experience, they are pretty useful and direct. Meanwhile, I am wondering if this batch of logs would cause some side effect of the SDK, for example, size?
Does Logger.debug or Logger.verbose help? Currently, we have infoWithObject and warnWithObject.

File: common4j/src/main/com/microsoft/identity/common/java/nativeauth/providers/NativeAuthResponseHandler.kt (hunk 3 above)
Comment (SammyO) [reply common4j/src/main/com/microsoft/identity/common/java/nativeauth/providers/NativeAuthResponseHandler.kt]: Yeah, let's remove this for now until we have a reason to put it back.

File: common4j/src/main/com/microsoft/identity/common/java/nativeauth/providers/interactors/ResetPasswordInteractor.kt (hunk 4 above)
Comment (SammyO) [reply common4j/src/main/com/microsoft/identity/common/java/nativeauth/providers/interactors/ResetPasswordInteractor.kt:121]: It won't affect the binary size, but might affect performance. I doubt it though, and I don't think we need to prematurely optimise for that.

**Comment by Yuki-YuXin** [comment]
Better to merge after multi access token work.

File: common4j/src/main/com/microsoft/identity/common/java/nativeauth/authorities/NativeAuthCIAMAuthority.kt [hunk 5]
```diff
@@ -104,8 +104,9 @@ class NativeAuthCIAMAuthority (
...
-        return (challengeTypes ?: emptyList()).plus(listOf(NativeAuthConstants.GrantType.REDIRECT)).distinct().joinToString(" ")
+        val challengeTypesWithDefault = (challengeTypes ?: emptyList()).plus(listOf(NativeAuthConstants.GrantType.REDIRECT)).distinct().joinToString(" ")
```
Comment (GBakolas) [inline common4j/src/main/com/microsoft/identity/common/java/nativeauth/authorities/NativeAuthCIAMAuthority.kt]: Nit: is it possible to simplify this expression, its a little confusing

File: common4j/src/main/com/microsoft/identity/common/java/nativeauth/authorities/NativeAuthCIAMAuthority.kt [hunk 6]
```diff
@@ -104,8 +104,9 @@ class NativeAuthCIAMAuthority (
...
+        Logger.info(TAG, "Challenge Types used = $challengeTypesWithDefault")
```
Comment (GBakolas) [inline common4j/src/main/com/microsoft/identity/common/java/nativeauth/authorities/NativeAuthCIAMAuthority.kt]: Missing brackets around challengeTypesWithDefault

File: common4j/src/main/com/microsoft/identity/common/java/nativeauth/authorities/NativeAuthCIAMAuthority.kt (hunk 5 above)
Comment (Yuki-YuXin) [reply common4j/src/main/com/microsoft/identity/common/java/nativeauth/authorities/NativeAuthCIAMAuthority.kt]: Use `challengeTypes.orEmpty() ` instead of `challengeTypes ?: emptyList()`. They are functionally equivalent. They returns an empty list if the original list is null.

File: common/src/main/java/com/microsoft/identity/common/nativeauth/internal/commands/AcquireTokenNoFixedScopesCommand.kt
Comment (GBakolas) [inline common/src/main/java/com/microsoft/identity/common/nativeauth/internal/commands/AcquireTokenNoFixedScopesCommand.kt:1]: why we have this empty file here? Probably this needs to be deleted

Comment (Yuki-YuXin) [reply common/src/main/java/com/microsoft/identity/common/nativeauth/internal/commands/AcquireTokenNoFixedScopesCommand.kt:1]: This file is deleted when multiple access token work. I think it's because of the merging update. Nice catch.

**Review by GBakolas — COMMENTED** [review]
Nit: In the ApiResponse files where you have removed the calls to logMethodCall() there are the LogSession import statements left. These statements are unused now so they can be removed for code cleanup

**Review by rmccahill — APPROVED** [review]

**Review by GBakolas — APPROVED** [review]
